router/src/infer.rs

```diff
@@ -523,18 +524,30 @@ fn send_responses(
     }
 
     // Create last Token
-    let token = Token {
-        id: generation.token_id,
-        text: generation.token_text,
-        logprob: generation.token_logprob,
-        special: generation.token_is_special,
+    let tokens: Vec<Token> = if let Some(tokens_) = generation.tokens {
+        tokens_
+            .ids
+            .into_iter()
+            .zip(tokens_.logprobs.into_iter())
+            .zip(tokens_.texts.into_iter())
+            .zip(tokens_.is_special.into_iter())
```

Comment:
`is_special` is empty for prompt tokens so this zip will not work.

Reply:
Prompt is handled in `Prefill` which doesn´t iter over them, so we're good.